### ISSUE TITLE
[Bug / Enhancement Request] When MinSize is used, but the font is smaller,then make sure that the text and buttons controls are vertically centered for "Up-Downers" (See DateTimePicker)

### ISSUE THREAD

**Issue by Smurf-IV** [Krypton-Suite/Standard-Toolkit#1653]
Related to #1651 , but this is specific to the "Up Downers"
> > They need to work for different "Base Fonts" (which just happen to be Segoe UI for V9)
>
> 3 - Is there a problem with that currently?
> 
3 - Yes, Still using a 9pt font but using "Small characters" then controls like the number up downer "Get confused":
![image](https://github.com/user-attachments/assets/72f142cb-d7a3-4cf1-bd97-b6e1ad2c6689)

The image shows what happens with the PR - Before it was rolled back. Now that is has been rolled back, the font and the buttons will be "Rammed" into the top of the control but because it's still a 9pt layout the "Height" of the control will still be the same !!!

**Comment by giduac** [comment]
Hi @Smurf-IV,

In the image the upper winforms nud also aligns the text upper left in the control.
So that is replicated in the Knud.

![Image](https://github.com/user-attachments/assets/0e1d20f0-d9e0-42c4-9719-308a3982baab)

In this example the Knud's inner control is vertically centered when the MinimumSize is set.

![Image](https://github.com/user-attachments/assets/3c6c3c7c-d7c3-45cf-8d28-64a706a9ca55)

So having it completely centred is off the table... ???

**Comment by Smurf-IV** [comment]
> So having it completely centred is off the table

I'm confused, Isn't your demo showing what you want ?

**Comment by giduac** [comment]
@Smurf-IV 

Compare the upper Winforms NUD to KNUD and look at the text alignment in the winforms control.

**Comment by Smurf-IV** [comment]
Is it the margin, and then pixel rounding (i.e. to sort of show a greater lower border area)?

**Comment by giduac** [comment]
the alignment is fixed upper/left....

**Comment by giduac** [comment]
@Smurf-IV 

Centring the inner control does not bring much to the table unless the outer canvas is bigger than the inner control.
Clearly the inner (winforms) control tends to align the text upper left on this smaller sized font and centres it better when the font size increases. Us centring the control wont bring much to the table.

I'm in favour of closing this and leave the control unchanged.

**Comment by Smurf-IV** [comment]
> I'm in favour of closing this and leave the control unchanged.

No, it is a problem, just a lower one than being able to see the text in all themes ;-)

**Comment by giduac** [comment]
> > I'm in favour of closing this and leave the control unchanged.
> 
> No, it is a problem, just a lower one than being able to see the text in all themes ;-)

Well, do you want the inner control to be centred on the canvas like here?
https://github.com/Krypton-Suite/Standard-Toolkit/issues/1653#issuecomment-2984569513

**Comment by Smurf-IV** [comment]
> Well, do you want the inner control to be centred on the canvas like here?

Again the confusion I'm seeing.. It already does that (Just not quite right IMO - hence my comments about it being a few pixels out).

So what is required?
Squashing into the top right, means that if you have a fatter border (To indicate focus or simulating finger press areas) then it would not look right.

**Comment by giduac** [comment]
@Smurf-IV 

A bit more digging:

For the winforms NUD
- When the font size is set to 14 the text is aligned in the middle
- When the size is below 14 aligned more towards the top
- when the size is over 14 aligned more towards the bottom.

- textbox has it too.
- maskedtextbox as well.

It looks like some kind of rounding offset probably the use of Math.Floor to calculate the top offset.
Lowercase text looks better aligned while uppercase show the offset. That's the same for numbers which appear like uppercase chars.

As the font grows towards "large" sizes this deficit is become less visible. Which underlines the possibility of rounding as the source of this.

**Comment by giduac** [comment]
@Smurf-IV 

And did a little custom drawing.

The rounding errors probably stem from the  internal use of TextRenderer.DrawText in combination with Graphics.MeasureString.

**Comment by Smurf-IV** [comment]
@giduac Good info, 
Should be able to use the float offsets now that screens are larger (With scaling etc) rather than rounding to the nearest pixel (like "we" had to do in the 640p days ;-))

**Comment by giduac** [comment]
@Smurf-IV 

It's a structural winforms "problem". 

Also I find it not necessarily problematic since the internal winforms controls have it / using the same winforms controls directly will expose the same behaviour.

.....

**Comment by Smurf-IV** [comment]
Ok, Closing..
